src/resolve/resolved-dns-query.c

```diff
@@ -500,6 +500,12 @@ DnsQuery *dns_query_free(DnsQuery *q) {
 
         free(q->request_address_string);
 
+        if (q->dnsservice_request)
+                dns_service_unref(q->dnsservice_request);
+
+        if (q->service_browser_request)
+                dns_service_browser_unref(q->service_browser_request);
```

Comment:
you can drop the conditionalizations. the unref calls should be fine with handling NULL